backend/crypt/cipher.go

```diff
@@ -114,22 +117,65 @@ func (mode NameEncryptionMode) String() (out string) {
 	return out
 }
 
-// Cipher defines an encoding and decoding cipher for the crypt backend
+// StrEncoding is encoding methods dealing with strings
+type StrEncoding interface {
```

Comment:
This should be called `filenameEncoding` - initial lower so it isn't exported and more specific name